api/handler/my_detail/update_email.go

```diff
@@ -0,0 +1,86 @@
+package mydetail
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"userland/api/helper"
+	"userland/api/response"
+	"userland/api/validator"
+	"userland/store"
+)
+
+type UpdateEmailRequest struct {
+	Email string `json:"email"`
+}
+
+func UpdateUserEmail(userStore store.UserStore, tokenStore store.TokenStore) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var request UpdateEmailRequest
+
+		userId, err := helper.AuthenticateUser(r, tokenStore)
+		if err != nil {
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(http.StatusUnauthorized)
+			json.NewEncoder(w).Encode(response.Unautorized_request(err.Error()))
+			return
+		}
+
+		err = json.NewDecoder(r.Body).Decode(&request)
+		if err != nil {
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(http.StatusBadRequest)
+			json.NewEncoder(w).Encode(response.Bad_request(err.Error()))
+			return
+		}
+
+		res, err := request.ValidateRequest()
+		if err != nil {
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(http.StatusUnprocessableEntity)
+			json.NewEncoder(w).Encode(response.UnproccesableEntity(res))
+			return
+		}
+
+		newEmail := store.User{
+			Email: request.Email,
+		}
+
+		err = userStore.UpdateUserEmail(r.Context(), newEmail, userId)
```

Comment:
Sure, that can do, force them re-login as soon as new email is verified.
However, still, send the verification code first, and ensure that the email is verified before do `UPDATE person SET email`. Otherwise, user can mistakenly insert the wrong email and lost the account.